src/linter/rules/seeded-randomness.ts

```diff
@@ -0,0 +1,140 @@
+	 * The default value for this is the function `set.seed` and the variable `.Random.seed`.
+	 */
+    randomnessProducers: {type: 'function' | 'assignment', name: string }[]
+	/**
+	 * A set of randomness consumer function names that require a seed to be set prior to invocation.
+	 */
+    randomnessConsumers: string[]
+}
+
+export interface SeededRandomnessMeta extends MergeableRecord {
+	consumerCalls:                 number
+	callsWithFunctionProducers:    number
+	callsWithAssignmentProducers:  number
+	callsWithNonConstantProducers: number
+}
+
+const assignments = DefaultBuiltinConfig.filter(b => b.type === 'function' && b.processor == 'builtin:assignment') as BuiltInFunctionDefinition<'builtin:assignment'>[];
+const assignmentSet = new Set<Identifier>(assignments.flatMap(b => b.names));
+const assignmentArgIndexes = new Map<string, number>(assignments.flatMap(a => a.names.map(n => ([n, a.config?.swapSourceAndTarget ? 1 : 0]))));
+
+export const SEEDED_RANDOMNESS = {
+	createSearch: (config) => Q.all()
+		.with(Enrichment.CallTargets, { onlyBuiltin: true })
+		.filter({
+			name: FlowrFilter.MatchesEnrichment,
+			args: {
+				enrichment: Enrichment.CallTargets,
+				test:       new RegExp(`"(.+:::?)?(${config.randomnessConsumers.join('|')})"`)
+			}
+		})
+		.with(Enrichment.LastCall,[
+			...config.randomnessProducers.filter(p => p.type === 'function').map(p => ({ callName: p.name })),
+			...[...assignmentSet].map(a => ({ callName: a, cascadeIf: CascadeAction.Continue }))
+		]),
+	processSearchResult: (elements, config, { dataflow }) => {
+		const assignmentProducers = new Set<string>(config.randomnessProducers.filter(p => p.type == 'assignment').map(p => p.name));
+		const metadata: SeededRandomnessMeta = {
+			consumerCalls:                 0,
+			callsWithFunctionProducers:    0,
+			callsWithAssignmentProducers:  0,
+			callsWithNonConstantProducers: 0
+		};
+		return {
+			results: elements.getElements()
+				// map and filter consumers
+				.flatMap(element => enrichmentContent(element, Enrichment.CallTargets).targets.map(target => {
+					metadata.consumerCalls++;
+					return {
+						range:         element.node.info.fullRange as SourceRange,
+						target:        target as Identifier,
+						searchElement: element
+					};
+				}))
+				// filter by calls that aren't preceded by a randomness producer
+				.filter(element => {
+					const producers = enrichmentContent(element.searchElement, Enrichment.LastCall).linkedIds
+						.map(e => dataflow.graph.getVertex(e.node.info.id) as DataflowGraphVertexFunctionCall);
+					const { assignment, func } = Object.groupBy(producers, f => assignmentSet.has(f.name) ? 'assignment' : 'func');
+					let nonConstant = false;
+
+					// function calls are already taken care of through the LastCall enrichment itself
+					for(const f of func ?? []) {
+						if(getValueOfArgument(dataflow.graph, f, { index: 0 }, [RType.Number, RType.String, RType.Logical]) !== undefined) {
```

Comment:
1) Shouldn't this go through a resolveArgument pass to check whether it is a variable that _points_ to a constant
2) the index should technically be configurable given that it may have a name etc.

Reply:
In general we probably want to "eval" the value to be future-proof against `set.seed(1042+2)` etc.